sphinx_cache/setup.py

```diff
@@ -0,0 +1,137 @@
+import os
+import shutil
+from pathlib import Path
+from typing import Any, Dict
+
+from sphinx.application import Sphinx
+from sphinx.config import Config
+
+from sphinx_cache.logging import show
+from sphinx_cache.utils import check_env_pickle_exist
+from sphinx_cache.version import __version__
+
+
+def setup(app: Sphinx) -> Dict[str, Any]:
+    # Define config values
+    app.add_config_value("cache_store_path", ".cache/", "html", types=[str])
+    app.add_config_value("cache_doctree_path", "_build/.doctrees", "html", types=[str])
+
+    # Make connections to events
+    app.connect("config-inited", config_init)
+    app.connect("builder-inited", build_init)
```

Comment:
I believe the first events get fired before Sphinx reads the cache because both `event.config-inited` and `event.builder-inited` are called before `event.doctree-read` according to the [sphinx-core-events](https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx-core-events) page.

Also, calling the `restore_cache()` function directly in the setup.py meant we could not access the extension's config values (i.e. cache_store_path and cache_doctree_path) and as a result, the `doctreedir` will still be the default one provided by the Sphinx HTML builder.